quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/AsyncPatternProcessorImpl.java

```diff
@@ -0,0 +1,107 @@
+import edu.asu.diging.quadriga.core.data.JobRepository;
+import edu.asu.diging.quadriga.core.exception.NodeNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.model.EventGraph;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.model.MappedTripleType;
+import edu.asu.diging.quadriga.core.model.events.pattern.PatternCreationEvent;
+import edu.asu.diging.quadriga.core.model.jobs.Job;
+import edu.asu.diging.quadriga.core.model.jobs.JobStatus;
+import edu.asu.diging.quadriga.core.service.AsyncPatternProcessor;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+import edu.asu.diging.quadriga.core.service.MappedTripleService;
+import edu.asu.diging.quadriga.core.service.PatternFinder;
+import edu.asu.diging.quadriga.core.service.PatternMapper;
+
+@Component
+public class AsyncPatternProcessorImpl implements AsyncPatternProcessor {
+
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private PatternFinder patternFinder;
+
+    @Autowired
+    private MappedTripleService mappedTripleService;
+
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+
+    @Autowired
+    private PatternMapper patternMapper;
+
+    @Autowired
+    private JobRepository jobRepository;
+
+    /* (non-Javadoc)
+     * @see edu.asu.diging.quadriga.core.service.AsyncPatternProcessor#processPattern(java.lang.String, java.lang.String, edu.asu.diging.quadriga.api.v1.model.PatternMapping, java.util.List)
+     */
+    @Async
+    @Override
+    public void processPattern(String jobId, String collectionId, PatternMapping patternMapping,
+            List<EventGraph> networks) {
+
+        Job job = jobRepository.findById(jobId).orElse(null);
+        PatternCreationEvent patternRoot = patternMapper.mapPattern(patternMapping);
+        
+        if (job == null || patternRoot == null) {
+            job.setStatus(JobStatus.FAILURE);
+            jobRepository.save(job);
+            return;
+        } else {
+            job.setStatus(JobStatus.PROCESSING);
+            jobRepository.save(job);
+        }
+        
+        MappedTripleGroup mappedTripleGroup;
+        try {
+            if (patternMapping.getMappedTripleGroupId() != null && !patternMapping.getMappedTripleGroupId().isEmpty()) {
+                mappedTripleGroup = mappedTripleGroupService.getByCollectionIdAndId(collectionId,
+                        patternMapping.getMappedTripleGroupId());
+                if (mappedTripleGroup == null) {
+                    job.setStatus(JobStatus.FAILURE);
+                    jobRepository.save(job);
+                    return;
+                }
+            } else {
+                mappedTripleGroup = mappedTripleGroupService.get(collectionId, MappedTripleType.DEFAULT_MAPPING);
+            }
```

Comment:
I"m confused what this if/else is for. The would be:
- a list of patterns and mappings is send to Quadriga
- a new mapped triple group is created
- all events that map one of the patterns are transformed according to the submitted mappings
triples should therefore never be added to the default mapping group. and the mapped triple groups that are being created would be of type custom mappings.

Reply:
I have created a new constant in MappedTripleType that is CUSTOM_MAPPING.